src/Selenium2Library/keywords/_element.py

```diff
@@ -636,7 +639,7 @@ def get_matching_xpath_count(self, xpath):
         `Xpath Should Match X Times`.
         """
         count = len(self._element_find("xpath=" + xpath, False, False))
-        return str(count)
+        return str(count) if return_str else count
```

Comment:
IMHO we should follow [DateTime's](https://github.com/robotframework/robotframework/blob/master/src/robot/libraries/DateTime.py) lead and use a util method to make sure that `str('false')` is treated as `False`, etc.
```
from robot.utils import is_truthy
...
return str(count) if is_truthy(return_str) else count
```